---
title: "assignment7"
author: "Zhihan Zhu"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
# import useful packages
suppressPackageStartupMessages({
  library(epiwraps)
  library(AnnotationHub)
  library(MotifDb)
  library(memes)
  library(universalmotif)
  library(ensembldb)
  library(ggplot2)
})

ah <- AnnotationHub(localHub=TRUE)
ensdb <- ah[["AH89211"]] # mouse ensembldb object
```

```{r}
# generate bw files we need
bam <- "D:/ETH/FS2024/Epigen/ethEpigen2024_materials/week07/atac.chr19.bam"

# the insertion (i.e. ‘cuts’) profile of nucleosome-free fragments
bam2bw(bam, output_bw = "NF_cuts.bw", paired=TRUE, binWidth=1L, type="ends", extend=2L, minFragLength=30, 
       maxFragLength=120, shift=c(4L,-5L), forceSeqlevelsStyle = "Ensembl")

# the centers of nucleosome-containing fragments
bam2bw(bam, output_bw = "NC_centers.bw", paired=TRUE, binWidth=5L, minFragLength=140, shift=c(4L,-5L),
       type="center", extent=10L, forceSeqlevelsStyle = "Ensembl")
```

```{r}
# get KLF4 motifs
motif_KLF4 <- MotifDb::query(MotifDb, c("KLF4","Mus"))[[1]]
motif2_KLF4 <- convert_motifs(motif_KLF4, class="TFBSTools-PFMatrix")

# get MAZ motifs
motif_MAZ <- MotifDb::query(MotifDb, c("MAZ","Mus"))[[1]]
motif2_MAZ <- convert_motifs(motif_MAZ, class="TFBSTools-PFMatrix")

# get the sequence for chr19
genome <- ah[["AH68356"]]
chr19 <- import(genome)["19"]

# find motif matches across chr19
moi_KLF4 <- motifmatchr::matchMotifs(motif2_KLF4, chr19, out = "positions", p.cutoff=1e-5)[[1]]
moi_KLF4 <- as(setNames(moi_KLF4, names(chr19)), "GRanges")

moi_MAZ <- motifmatchr::matchMotifs(motif2_MAZ, chr19, out = "positions", p.cutoff=1e-5)[[1]]
moi_MAZ <- as(setNames(moi_MAZ, names(chr19)), "GRanges")
```

```{r}
# prepare the list of tracks
tracks <- c("NF cuts"="NF_cuts.bw", "NC centers"="NC_centers.bw")

# extract signals around motif occurences
sm_KLF4 <- signal2Matrix(tracks, moi_KLF4, w=5, extend=1000)
sm_MAZ <- signal2Matrix(tracks, moi_MAZ, w=5, extend=1000)

# use background normalization
nf <- getNormFactors(tracks, useSeqLevels = "19", nwind = 5000L)
sm_KLF4 <- renormalizeSignalMatrices(sm_KLF4, scaleFactors = nf)
sm_MAZ <- renormalizeSignalMatrices(sm_MAZ, scaleFactors = nf)
```

```{r}
# plot the signals for KLF4
plotEnrichedHeatmaps(sm_KLF4, trim = 0.95, colors = c("white", "darkred"))
```

```{r}
# plot the signals for MAZ
plotEnrichedHeatmaps(sm_MAZ, trim = 0.95, colors = c("white", "darkred"))
```

