---
title: "assignment6"
author: "Zhihan Zhu"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
# import libraries
suppressPackageStartupMessages({
  library(BiocIO)
  library(AnnotationHub)
  library(ensembldb)
  library(GenomicRanges)
  library(ggplot2)
  library(memes) # for the meme-based methods -- COMMENT OUT when using alternatives
  library(motifmatchr) # for scanning sequences for matches of given motifs
  library(Biostrings) # for handling sequences
  library(MotifDb) # database of motifs
  library(TFBSTools) # for handling some motif formats
  library(universalmotif) # for converting motifs to various formats
  library(PWMEnrich) # for R-based motif enrichment analysis
})
```


```{r}
# download file
download.file("https://www.encodeproject.org/files/ENCFF403FJH/@@download/ENCFF403FJH.bed.gz", "D:/ETH/FS2024/Epigen/data/week6/WTC11_CREB1_ENCFF403FJH.bed.gz")
```

```{r}
peaks <- rtracklayer::import("D:/ETH/FS2024/Epigen/data/week6/WTC11_CREB1_ENCFF403FJH.bed.gz", format="NarrowPeak")
seqlevelsStyle(peaks) <- "Ensembl"
# look at chromosome 1
peaks_chr1 <- peaks[seqnames(peaks)=="1"]
```

### identify the instance of the factor's motif

```{r}
# search in motif database
motifs <- query(MotifDb, "CREB1")
names(motifs)
```

```{r}
motif <- motifs[["Hsapiens-HOCOMOCOv10-CREB1_HUMAN.H10MO.A"]]
view_motifs(motif)
```

```{r}
# get the genome sequence
ah <- AnnotationHub()
AnnotationHub::query(ah, c("ensembl", "Homo sapiens", "2bit"))
```

```{r}
genome <- ah[["AH106285"]]
genome_seqs <- import(genome)
```

```{r}
Biostrings::writeXStringSet(genome_seqs, "D:/ETH/FS2024/Epigen/data/week6/genome.fa")
```

```{r}
motif2 <- convert_motifs(motif, class="TFBSTools-PWMatrix")
moi <- motifmatchr::matchMotifs(motif2, subject=peaks_chr1, genome=Rsamtools::FaFile("D:/ETH/FS2024/Epigen/data/week6/genome.fa"), out="positions")
moi <- moi[[1]] # we scanned for just one motif, so we get the results for that motif
head(moi)
```

```{r}
# Of all the peaks, what proportion contains a motif for the factor?
table(overlapsAny(peaks_chr1, moi))
```

Of 500 peaks (of chromosome 1), 36 (7.2%) contain a motif.

```{r}
# scan the whole genome
motif_across_genome <- matchMotifs(motif2, subject=genome_seqs, out="positions")[[1]]
# when inputing matchMotifs directly with sequences, one has to do the following 
# to transform the output to GRanges:
names(motif_across_genome) <- names(genome_seqs)
motif_across_genome <- as(motif_across_genome, "GRanges")
head(motif_across_genome)
```

```{r}
table(overlapsAny(motif_across_genome, peaks))
```

Of the 118363 motif instances, 476 (0.4%) overlap a peak.